### PR TITLE
Using new regex for path validation

### PR DESCRIPTION
**Pull request by AdamMagaluk**
From: https://github.com/30x/enrober/pull/123/commits/654a7a377fa719f3e4ae4733b6d77abd4a654f2a

### CODE REVIEW

File: router/pods_test.go [hunk 1]
```diff
@@ -269,6 +269,54 @@ func TestGetRoutesInvalidPublicPathsPath(t *testing.T) {
...
+				},
+			},
+		},
+		Status: api.PodStatus{
+			Phase: api.PodRunning,
+			PodIP: "1.2.3.4",
+		},
+	}))
+
+	// "%ZZ" is not a valid path for targetPath segment
```
Comment (whitlockjc) [inline router/pods_test.go]: Comment does not match the test.

File: router/pods_test.go [hunk 2]
```diff
@@ -269,6 +269,54 @@ func TestGetRoutesInvalidPublicPathsPath(t *testing.T) {
...
+	validateRoutes(t, "pod has an invalid routingPaths path", []*Route{}, GetRoutes(config, &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Annotations: map[string]string{
+				config.PodsPathsAnnotation: genRoutes(path("/test/%2a/%", "3000", "")),
```
Comment (whitlockjc) [inline router/pods_test.go]: Do you mind adding an example path with bad and good encoding in the middle of the string?  The issues being fixed by the new regex are:

1. Previous regex only ensured one match was found _(not that the full string matched)_
2. Previous regex only matched either a valid character _(singluar)_ or a valid URI encoded character _(singular)_

This test would not show that the regex was wrong.

Comment (AdamMagaluk) [inline router/pods_test.go]: See updates, has separate unit test for `validatePaths` that tests all three cases same as enrober.

**Comment by whitlockjc** [comment]
We still need a test where we mix unencoded URI characters _(`A-Za-z0-9\-._~!$&'()*+,;=:@`)_ and encoded characters _(`%[0-9A-Fa-f]{2}`)_ in the same path segment.  Reason being is our previous regex would match things improperly and was not guaranteed to fail whenever a passing character had been found.  For example, it would catch the failure `/%zz` but would not `/hello%zzworld` failure.  Make sense?

**Comment by whitlockjc** [comment]
LGTM

**Pull request merged**